lib/rules/lint-no-trailing-spaces.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const Rule = require('./base');
+
+module.exports = class NoTrailingSpaces extends Rule {
+  visitor() {
+    return {
+
+      // implementation goes here in exit(): in the exit handler, the rule will not
+      // be called if it has been disabled by any inline comments within the file.
+
+      Program(node) {
+        let line = node.loc.start.line;
+        let column = node.loc.start.column;
+
+        // yielded content makes it in here too
+        //
+        // `{{#my-component}}
+        //   test
+        // {{/my-component}}`
+        //
+        // becomes
+        //
+        // `
+        //   test
+        // `
+        //
+        // check for that case
+        if (line !== 1 || column !== 0) {
+          return;
+        }
+
+        let source = this.sourceForNode(node);
+
+        source.split('\n').forEach((line, i) => {
```

Comment:
The lines are already available in `this.source` as an array of lines for the whole file.

Reply:
it includes the newlines `[ '{{#my-component}}\n', '  test \n', '{{/my-component}}', '' ]`, which means it wouldn't be as clean as a simple forEach. Would you still prefer I switch?